magento/pwa-studio
# PWA-2783: Track Search Requests

- Added search event dispatch
- Added test for change



## Description

Overview

As a merchant, I want to be able to track search requests closely to monitor users behavior.

Preconditions
Framework with a example extension created
Acceptance Criteria
1. SearchRequestSent event is tracked every time user initiates a product search with the attributes that can be mapped to the following schema

refinements: (attribute: string; value: string; isRange: boolean )
sort (attribute: string; order: string )
query: string;
2. Example extension subscribes to the events and has events available for processing

Example extension simply subscribes to event and logs its occurrence


## Related Issue







Closes #PWA-2783

## Acceptance





### Verification Stakeholders



### Specification




## Verification Steps






#### Test scenario(s) for direct fix/feature





Test plan
1. Perform a product search
2. Open browser console and verify that the sample extension logged the search request event (with its contextual data)
3. Refine the search using one of the available attributes
4. Open browser console and verify that the sample extension logged the search request event with the refinement data
5. Sort the search results
6. Open browser console and verify that the sample extension logged the search request event with the sort data

#### Test scenario(s) for any existing impacted features/areas





#### Test scenario(s) for any Magento Backend Supported Configurations





#### Is Browser/Device testing needed?





#### Any ad-hoc/edge case scenarios that need to be considered?




## Screenshots / Screen Captures (if appropriate)

## Breaking Changes (if any)





## Checklist



-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.
